firebase_admin/remote_config.py

```diff
@@ -0,0 +1,225 @@
+"""
+
+import json
+from typing import Any, Dict, Optional
+import requests
+from firebase_admin import App, _http_client, _utils
+import firebase_admin
+
+_REMOTE_CONFIG_ATTRIBUTE = '_remoteconfig'
+
+class ServerTemplateData:
+    """Represents a Server Template Data class."""
+    def __init__(self, etag, template_data):
+        """Initializes a new ServerTemplateData instance.
+
+        Args:
+            etag: The string to be used for initialize the ETag property.
+            template_data: The data to be parsed for getting the parameters and conditions.
+
+        Raises:
+            ValueError: If the template data is not valid.
+        """
+        if 'parameters' in template_data:
+            if template_data['parameters'] is not None:
+                self._parameters = template_data['parameters']
+            else:
+                raise ValueError('Remote Config parameters must be a non-null object')
+        else:
+            self._parameters = {}
+
+        if 'conditions' in template_data:
+            if template_data['conditions'] is not None:
+                self._conditions = template_data['conditions']
+            else:
+                raise ValueError('Remote Config conditions must be a non-null object')
+        else:
+            self._conditions = []
+
+        self._version = ''
+        if 'version' in template_data:
+            self._version = template_data['version']
+
+        self._etag = ''
+        if etag is not None and isinstance(etag, str):
+            self._etag = etag
+
+    @property
+    def parameters(self):
+        return self._parameters
+
+    @property
+    def etag(self):
+        return self._etag
+
+    @property
+    def version(self):
+        return self._version
+
+    @property
+    def conditions(self):
+        return self._conditions
+
+
+class ServerTemplate:
+    """Represents a Server Template with implementations for loading and evaluting the tempalte."""
+    def __init__(self, app: App = None, default_config: Optional[Dict[str, str]] = None):
+        """Initializes a ServerTemplate instance.
+
+        Args:
+          app: App instance to be used. This is optional and the default app instance will
+                be used if not present.
+          default_config: The default config to be used in the evaluated config.
+        """
+        self._rc_service = _utils.get_app_service(app,
+                                                  _REMOTE_CONFIG_ATTRIBUTE, _RemoteConfigService)
+
+        # This gets set when the template is
+        # fetched from RC servers via the load API, or via the set API.
+        self._cache = None
+        if default_config is not None:
+            self._stringified_default_config = json.dumps(default_config)
+        else:
+            self._stringified_default_config = None
+
+    async def load(self):
+        """Fetches the server template and caches the data."""
+        self._cache = self._rc_service.get_server_template()
```

Comment:
Thanks for the suggestion! This coroutine will be block the event loop during the network call. As discussed on the other comment, the idea was to run this coroutine inside `asyncio.run(..)` to wait for the routine to finish, but the event loop would still be blocked.

I updated the `internal get_server_template method` to be async as well and it will have the blocking network call `inside the loop.run_in_executor(...)` as you suggested to ensure that the event loop is not blocked.